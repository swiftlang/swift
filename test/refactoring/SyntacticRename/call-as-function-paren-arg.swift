struct Adder {
    var base: Int
    func /*test:def*/callAsFunction(_ x: Int) -> Int {
        return base + x
    }
}

let add3 = Adder(base: 3)
_ = add3/*test:call*/(10)
_ = add3 . /*test:call*/callAsFunction(10)
_ = add3 . /*test:ref*/callAsFunction(_:)
_ = add3 . /*test:ref*/callAsFunction

// REQUIRES: swift_swift_parser
// RUN: %empty-directory(%t.ranges)
// RUN: %refactor -find-rename-ranges -source-filename %s -pos="test" -is-function-like -old-name "callAsFunction(_:)" >> %t.ranges/call-as-function-paren-arg.swift.expected
// RUN: diff -u %S/Outputs/call-as-function-paren-arg.swift.expected %t.ranges/call-as-function-paren-arg.swift.expected
