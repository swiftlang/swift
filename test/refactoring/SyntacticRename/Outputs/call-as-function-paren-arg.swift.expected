struct Adder {
    var base: Int
    func /*test:def*/<keywordBase>callAsFunction</keywordBase>(<arglabel index=0>_</arglabel><param index=0> x</param>: Int) -> Int {
        return base + x
    }
}

let add3 = Adder(base: 3)
_ = add3/*test:call*/(<callcombo index=0></callcombo>10)
_ = add3 . /*test:call*/<keywordBase>callAsFunction</keywordBase>(<callcombo index=0></callcombo>10)
_ = add3 . /*test:ref*/<keywordBase>callAsFunction</keywordBase>(<sel index=0>_</sel>:)
_ = add3 . /*test:ref*/<keywordBase>callAsFunction</keywordBase>

